services/validation.ts

```diff
@@ -1,24 +1,42 @@
 import axios from 'axios';
-import { getCookie } from 'cookies-next';
+import { getCookie, setCookie } from 'cookies-next';
 import { api } from './url';
 
 export async function verifyToken() {
-    const token = getCookie('access_token');
+    const accessToken = getCookie('access_token');
+    const refreshToken = getCookie('refresh_token');
     const url = `${api.portal.backend.baseUrl}${api.portal.backend.verify}`;
+    const refreshUrl = `${api.portal.backend.baseUrl}${api.portal.backend.refreshToken}`;
 
-    if (!token) {
-        return { isValid: false, message: 'Token not found' };
+    if (!accessToken) {
+        return { isValid: false, message: 'Access token not found' };
     }
 
     try {
         const response = await axios.get(url, {
             headers: {
-                Authorization: `Bearer ${token}`,
+                Authorization: `Bearer ${accessToken}`,
             },
         });
+
         return { isValid: true, data: response.data };
-        
-    } catch (error) {
+    } catch (error: any) {
+        if (error.response.data.detail === "Signature has expired" && refreshToken) {
+            try {
+                const refreshResponse = await axios.post(refreshUrl, {}, {
+                    headers: {
+                        Authorization: `Bearer ${refreshToken}`,
+                    },
+                });
+
+                setCookie('access_token', refreshResponse.data.access_token);
+
+                return { isValid: true };
+            } catch (refreshError) {
+                return { isValid: false, message: 'Error refreshing token', refreshError };
+            }
+        }
+
         return { isValid: false, message: error };
```

Comment:
I didn't see the code to redirect to portal (with params set for DelhiStudents) if the token is missing. Is it included here?

Reply:
It's there in `AuthContext.tsx`. Line 38